src/platforms/ios/iosiaphandler.mm

```diff
@@ -198,88 +197,45 @@ - (void)request:(SKRequest*)request didFailWithError:(NSError*)error {
 
 @end
 
-// static
-IAPHandler* IAPHandler::createInstance() {
-  Q_ASSERT(!s_instance);
-  new IAPHandler(qApp);
-  Q_ASSERT(s_instance);
-  return instance();
-}
-
-// static
-IAPHandler* IAPHandler::instance() {
-  Q_ASSERT(s_instance);
-  return s_instance;
-}
-
-IAPHandler::IAPHandler(QObject* parent) : QAbstractListModel(parent) {
-  MVPN_COUNT_CTOR(IAPHandler);
+IOSIAPHandler::IOSIAPHandler(QObject* parent) : IAPHandler(parent) {
+  MVPN_COUNT_CTOR(IOSIAPHandler);
 
   Q_ASSERT(!s_instance);
   s_instance = this;
 
-  m_delegate = [[IAPHandlerDelegate alloc] initWithObject:this];
+  m_delegate = [[IOSIAPHandlerDelegate alloc] initWithObject:this];
   [[SKPaymentQueue defaultQueue]
-      addTransactionObserver:static_cast<IAPHandlerDelegate*>(m_delegate)];
+      addTransactionObserver:static_cast<IOSIAPHandlerDelegate*>(m_delegate)];
 }
 
-IAPHandler::~IAPHandler() {
-  MVPN_COUNT_DTOR(IAPHandler);
+IOSIAPHandler::~IOSIAPHandler() {
+  MVPN_COUNT_DTOR(IOSIAPHandler);
 
   Q_ASSERT(s_instance == this);
   s_instance = nullptr;
 
-  IAPHandlerDelegate* delegate = static_cast<IAPHandlerDelegate*>(m_delegate);
+  IOSIAPHandlerDelegate* delegate = static_cast<IOSIAPHandlerDelegate*>(m_delegate);
   [[SKPaymentQueue defaultQueue] removeTransactionObserver:delegate];
 
   [delegate dealloc];
   m_delegate = nullptr;
 }
 
-void IAPHandler::registerProducts(const QByteArray& data) {
-  logger.debug() << "Maybe register products";
-
-  Q_ASSERT(m_productsRegistrationState == eRegistered ||
-           m_productsRegistrationState == eNotRegistered);
-
-  auto guard = qScopeGuard([&] { emit productsRegistered(); });
-
-  if (m_productsRegistrationState == eRegistered) {
-    return;
-  }
-
-  Q_ASSERT(m_products.isEmpty());
-
-  QJsonDocument json = QJsonDocument::fromJson(data);
-  if (!json.isObject()) {
-    logger.debug() << "Object expected";
-    return;
-  }
-
-  QJsonObject obj = json.object();
-  if (!obj.contains("products")) {
-    logger.debug() << "products entry expected";
-    return;
-  }
-
-  QJsonArray products = obj["products"].toArray();
-  if (products.isEmpty()) {
-    logger.error() << "No products found";
-    return;
-  }
-
-  m_productsRegistrationState = eRegistering;
-
-  for (const QJsonValue& value : products) {
```

Comment:
when i moved this line to cpp file the clang compiler complained in ci (although not on my machine), so i changed it to `auto` - added a comment at relevant line above.

Reply:
what's the error?

Reply:
https://github.com/mozilla-mobile/mozilla-vpn-client/runs/3270943125#step:10:1389

Reply:
ah! right! The reason why we don't see it in the mm file is that... we use a "different" compiler. The fix is: `for (const QJsonValue value : product) {` or without `const`.